mrp_stock_analytic/tests/test_mrp_stock_analytic.py

```diff
@@ -0,0 +1,148 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.tests import Form
+
+from odoo.addons.stock_analytic.tests.test_stock_picking import TestStockPicking
+
+
+class TestMrpStockAnalytic(TestStockPicking):
```

Comment:
@aisopuro This is not correct.

As the imported class will launch again tests from stock_analytic module.

Either you don't depend on it (TransactionCase instead), either you should first do a PR (we can quickly merge it) to do a common class for the setup.

Reply:
Good find! It *can* be a valid approach to ensure that the original tests still work after installation of this module. Is that what was intended @yostashiro?
If not, then maybe the extraction of a common setup class can still be included in this PR (if in a separate commit)?

Reply:
Thanks for the help @rousseldenis . Opened #682 and marked this as depending on it

Reply:
Oh, seems it didn't fix it :disappointed: . Looks like this only shows up when running the at-install tests in a fresh database. I don't get the error on a second run in an already installed database.

Reply:
In fact, this is partly better. The error is no more launched by mrp_stock_analytic tests but in stock_analytic only

Reply:
@yostashiro Thanks for clarifying!

Reply:
I think I've tracked down the cause of the issue, but I'm not sure how to fix/avoid it.

Symptoms:
* When I install a fresh database and install `stock_analytic`, its tests pass
* When I install a fresh database and install `mrp_stock_analytic` and run the tests for `stock_analytic` its tests fail

Based on debugging, the issue seems to be that `mrp_stock_analytic` depends on both `mrp` and `stock` (as it should), and that makes Odoo load them in the order
1. `stock`
2. `mrp`
3. `stock_analytic`
4. `mrp_stock_analytic`

When `mrp` is loaded, it contains demo data that sets the default resupply on the test warehouse to happen through manufacture rather than shipping in new product through pickings: https://github.com/odoo/odoo/blob/a2909105db3e016bfbbd092c9c10498dc610cfc1/addons/mrp/data/mrp_data.xml#L18

This then replaces/deactivates the routes in such a way that the normal stock operations no longer work. I've tried a couple things (enabling the manufacturing route on the product, disabling the option the `mrp` module activates), but I either get the same error, or the analytics aren't propagated when the product is manufactured, because of course not, that is the job of `mrp_stock_analytic`, which is not yet loaded.

Any ideas for how to solve this issue would be appreciated.

Reply:
Solved this, those interested can check https://github.com/OCA/account-analytic/pull/644/commits/cb3995423905eb12c1a652d314133c7791e5e235 for details.